zkevm-circuits/src/sha256_circuit/circuit.rs

```diff
@@ -0,0 +1,1249 @@
+use halo2_gadgets::sha256::{table16::*, Sha256Instructions, BLOCK_SIZE};
+use halo2_proofs::{
+    circuit::{AssignedCell, Layouter, Region, Value},
+    halo2curves::bn256::Fr,
+    plonk::{
+        Advice, Any, Column, ConstraintSystem, Constraints, Error, Expression, Fixed, Selector,
+        TableColumn,
+    },
+    poly::Rotation,
+};
+use itertools::Itertools;
+use std::convert::TryInto;
+type BlockState = <Table16Chip as Sha256Instructions<Fr>>::State;
+
+/// u32 size for SHA256 digit
+pub const DIGEST_SIZE: usize = 8;
+
+/// the defination for a sha256 table
+pub trait SHA256Table {
+    /// the cols has layout [s_enable, input_bytes, input_len, hashes, effect]
+    fn cols(&self) -> [Column<Any>; 5];
+
+    /// ...
```

Comment:
Generally we should avoid adding `/// ...` simply to fix `cargo doc` warnings/failures. It should take not much time to add a small doc regarding the method. Its a nitpick but it helps a lot our code quality and readability.

Reply:
done